src/env.ts

```diff
@@ -1,3 +1,31 @@
 import dotenv from 'dotenv'
 
-dotenv.config()
+import type { Env } from './types'
+
+const prepareEnv = () => {
+  dotenv.config()
+
+  return {
+    ...process.env,
+
+    CI_MERGE_REQUEST_IID: +process.env.CI_MERGE_REQUEST_IID!,
+    GITLAB_CI_USER_EMAIL:
+      process.env.GITLAB_CI_USER_EMAIL ||
+      'gitlab[bot]@users.noreply.gitlab.com',
+    GITLAB_COMMENT_TYPE: process.env.GITLAB_COMMENT_TYPE ?? 'discussion',
+    DEBUG_GITLAB_CREDENTIAL: process.env.DEBUG_GITLAB_CREDENTIAL ?? 'false',
+
+    // only check for the token if we are explicitly using it
+    // eslint-disable-next-line sonar/function-name
+    get GITLAB_TOKEN() {
```

Comment:
Is a getter really needed?

Reply:
Without using a getter, it will validate that even when running in the CI pipeline in this repository and thus make the pipelines fail. 😅

Reply:
Would calling `setFailed` multiple times be problematic? It is likely to be accessed two times.

Reply:
We can cache the result then.